### PR TITLE
Add a GCP bastion module, move the AWS module to a sub-directory, use a Docker image for `terraform-docs`

### PR DESCRIPTION
**Pull request by ivanfetch**
The new GCP Terraform module manages a Google Cloud bastion, using Google DNS for the bastion host record, Stackdriver for logs, and Google Cloud Storage for SSH hostkeys and optional `additional external users`.

The AWS bastion module is moved to its own `aws` sub-directory - the next release of the AWS module will require users to update their `source` argument to include the `//aws` path and a module-specific release tag such as `aws-vx.y.z`.

A Docker image is used to incorparate `terraform-docs` output into per-module ReadMe files, until a version of terraform-docs is released that natively supports Terraform 0.12.
* Terraform-docs pull request: https://github.com/segmentio/terraform-docs/pull/113
* Docker image and Terraform 0.12 workaround: https://github.com/cytopia/docker-terraform-docs/pull/8

### CODE REVIEW

File: gcp/example-usage [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  infrastructure_bucket_bastion_key = "default/bastion"
+
+  dns_zone_name = "kube-company-com"
+
+  # This network houses the subnetwork below it.
+  network_name    = "kube"
+  subnetwork_name = "kube-nodes"
+
+  # This uses an SSH public key from the Terraform root module directory.
+  ssh_public_key_file                = file("${path.root}/bastion_ssh.pub")
```
Comment (dosullivan) [inline gcp/example-usage]: using the `file` function as shown here clashes with the `file` function in `instance-template.tf`. It worked if I dropped the function here, so the example should probably be updated to `ssh_public_key_file                = "${path.root}/../config/private/${var.ssh_key}"`

File: gcp/example-usage [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  unattended_upgrade_email_recipient = "you@example.com"
+
+  # Execute additional commands during User Data, for this bastion.
+  additional_user_data = <<EOD
```
[the remark this answers is not in the document]
Comment (dosullivan) [reply gcp/example-usage]: Just read further into the GCP module docs. It looks like `additional_user_data` should be replaced with `additional_setup_script` here.

[the remark this answers is not in the document]
Comment (ivanfetch) [reply gcp/example-usage]: Yes - thank you - fixed as part of work earlier toady.

File: gcp/example-usage (hunk 1 above)
Comment (ivanfetch) [reply gcp/example-usage]: I gave this input an unfortunate name in the AWS module - I've updated the GCP module to match the AWS one, and updated the input descriptions to clarify how the input is used despite its name. The input represents the content, not the file name, of the SSH public key. I opened [this issue](https://github.com/FairwindsOps/terraform-bastion/issues/25) to consider renaming the input.

Comment (dosullivan) [reply gcp/example-usage]: Ah ok, thanks for clarifying!

**Comment by ejether** [comment]
Re-requested review from @dosullivan to ensure his concerns were addressed.

**Comment by ejether** [comment]
@ivanfetch 
Am I correct in thinking this new sub-directory structure will force existing aws `bastion.tf` files to change their source from 
` source  = "git::https://github.com/fairwindsops/terraform-bastion.git?ref=v0.4.1"`
to 
` source  = "git::https://github.com/fairwindsops/terraform-bastion.git/aws?ref=v0.4.1"`

or some rough equivalent?

**Comment by ejether** [comment]
@ivanfetch have you had a chance to checkout out my question above?

**Comment by ivanfetch** [comment]
> @ivanfetch have you had a chance to checkout out my question above?

**Comment by ivanfetch** [comment]
> @ivanfetch
> Am I correct in thinking this new sub-directory structure will force existing aws `bastion.tf` files to change their source from
> ` source = "git::https://github.com/fairwindsops/terraform-bastion.git?ref=v0.4.1"`
> to
> ` source = "git::https://github.com/fairwindsops/terraform-bastion.git/aws?ref=v0.4.1"`
> 
> or some rough equivalent?

@ejether Correct, although not until the next release which tags the AWS-specific module. I’ve already updated the changelog to reflect what you described above, which won’t take affect until the next AWS module release.

**Comment by ejether** [comment]
Great, thanks!

**Pull request merged**